src/test/java/bio/terra/service/snapshot/SnapshotConnectedTest.java

```diff
@@ -162,29 +172,56 @@ public void testArrayStruct() throws Exception {
 
   @Test
   public void testEnumeration() throws Exception {
-    SnapshotRequestModel snapshotRequest =
-        SnapshotConnectedTestUtils.makeSnapshotTestRequest(
-            jsonLoader, datasetSummary, "snapshot-test-snapshot.json");
+    datasetDao.patch(datasetSummary.getId(), new DatasetPatchRequestModel().phsId(PHS_ID));
 
     // Other unit tests exercise the array bounds, so here we don't fuss with that here.
     // Just make sure we get the same snapshot summary that we made.
     List<SnapshotSummaryModel> snapshotList = new ArrayList<>();
-    for (int i = 0; i < 5; i++) {
+
+    // A snapshot is authorized to be included in enumeration if the caller can access it directly
+    // via SAM and/or indirectly via a linked RAS passport.
+    // Create three snapshots with consent code -- authorized indirectly via a linked RAS passport.
+    SnapshotRequestModel rasSnapshotRequest =
+        SnapshotConnectedTestUtils.makeSnapshotTestRequest(
+                jsonLoader, datasetSummary, "snapshot-test-snapshot.json")
+            .consentCode(CONSENT_CODE);
+    List<UUID> rasSnapshotIds = new ArrayList<>();
+    for (int i = 0; i < 3; i++) {
+      MockHttpServletResponse response = performCreateSnapshot(rasSnapshotRequest, "_en_");
+      SnapshotSummaryModel summaryModel = validateSnapshotCreated(rasSnapshotRequest, response);
+      snapshotList.add(summaryModel);
+      rasSnapshotIds.add(summaryModel.getId());
+    }
+    assertThat("3 RAS-authorized snapshots", rasSnapshotIds.size(), equalTo(3));
```

Comment:
There's a matcher for collection size
```suggestion
    assertThat("3 RAS-authorized snapshots", rasSnapshotIds, hasSize(3));
```